plugins/osdn/node.go

```diff
@@ -0,0 +1,140 @@
+package osdn
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	log "github.com/golang/glog"
+
+	"github.com/openshift/openshift-sdn/pkg/netutils"
+	"github.com/openshift/openshift-sdn/plugins/osdn/api"
+
+	osclient "github.com/openshift/origin/pkg/client"
+	osapi "github.com/openshift/origin/pkg/sdn/api"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	kubeletTypes "k8s.io/kubernetes/pkg/kubelet/container"
+	kexec "k8s.io/kubernetes/pkg/util/exec"
+	kubeutilnet "k8s.io/kubernetes/pkg/util/net"
+)
+
+type OsdnNode struct {
+	multitenant        bool
+	registry           *Registry
+	localIP            string
+	localSubnet        *osapi.HostSubnet
+	hostName           string
+	podNetworkReady    chan struct{}
+	vnids              vnidMap
+	iptablesSyncPeriod time.Duration
+	mtu                uint
+}
+
+// Called by higher layers to create the plugin SDN node instance
+func NewNodePlugin(pluginName string, osClient *osclient.Client, kClient *kclient.Client, hostname string, selfIP string, iptablesSyncPeriod time.Duration, mtu uint) (api.OsdnNodePlugin, error) {
```

Comment:
We could... but look at the code in origin: https://github.com/openshift/origin/pull/9476/files#diff-232120b6845c4d46e68c543a21dbf8c8L377. The master plugin object didn't get used anyway, origin just created it, called Start() on it, and then forgot about it.

I think with a bit of further rewriting we should eventually be able to similarly squash the node and proxy code too; right now they're created and started in different files, but I don't think that's actually necessary.